framework/helpers/BaseUrl.php

```diff
@@ -83,12 +83,13 @@ class BaseUrl
      *
      * @param string|array $route use a string to represent a route (e.g. `index`, `site/index`),
      * or an array to represent a route with query parameters (e.g. `['site/index', 'param1' => 'value1']`).
-     * @param bool|string $scheme the URI scheme to use in the generated URL:
+     * @param null|bool|string $scheme the URI scheme to use in the generated URL:
      *
      * - `false` (default): generating a relative URL.
      * - `true`: returning an absolute base URL whose scheme is the same as that in [[\yii\web\UrlManager::$hostInfo]].
      * - string: generating an absolute URL with the specified scheme (either `http`, `https` or empty string
      *   for protocol-relative URL).
+     * - `null`: generating an absolute URL with the current request scheme.
```

Comment:
How this differ from `returning an absolute base URL whose scheme is the same as that in [[\yii\web\UrlManager::$hostInfo]].`?

Reply:
@rob006 can you give an example please?

Reply:
AFAIK `null` works the same as `true`, so what is the point of documenting this? If you pass an object or array it will also work as `true`, but there is no sense to document every possible variation for this param.

Reply:
So... Of course, you right.

But `true` antagonist of `null` => we have flip logic. Some strange.
I expect to use same value in each way to create full link:

```php
foreach (['http', 'https', '', null] as $scheme){
  \yii\helpers\Url::toRoute(['/user/view', 'id' => $id], $scheme);
  \Yii::$app->urlManager->createAbsoluteUrl(['/user/view', 'id' => $id], $scheme);
}
```

Reply:
additional, avoid of doc in createAbsoluteUrl is same with true
```php
foreach (['http', 'https', '', null, true] as $scheme){
  \yii\helpers\Url::toRoute(['/user/view', 'id' => $id], $scheme);
  \Yii::$app->urlManager->createAbsoluteUrl(['/user/view', 'id' => $id], $scheme);
}
```

Reply:
Can we change it like this:
```php
     * - `true` | `null`: returning an absolute base URL whose scheme is the same as that in [[\yii\web\UrlManager::$hostInfo]].
```

Reply:
>  I expect to use same value in each way to create full link:

Why? These are different components, they works differently, why do you expect similar behavior? Documenting `null` as possible value will only complicate API of this method.

Reply:
Just because I expected easy for understand clearly design.

Anyway, current realization is correct.